### DISCUSSION TITLE
why am i getting "WARNING:adapters.loading:Overwriting existing adapter 'adapter_name'." after training finishes?

### DISCUSSION THREAD

**Discussion by MRawhani** [adapter-hub/adapters#710] [Q&A] [answered]
Whenever I use AdapterTrainer object for training i get this message when training finishes:
"WARNING:adapters.loading:Overwriting existing adapter 'adapter_name'."
see this sample:
![image](https://github.com/adapter-hub/adapters/assets/37050593/047dc50e-9f71-412c-9055-d6d56d8706cc)

whatever task i am using it gives me this warning, why? and does it effect the performance?

**Comment by calpt** [comment, marked as answer]
Hi @MRawhani, this is expected if you use something like `load_best_model_at_end` for your training since it will re-load the best adapter checkpoint after training has finished. In that case it does not affect performance.

We'll look into making this message less confusing in the future.
